endpoint/endpoint.go

```diff
@@ -26,6 +26,14 @@ const (
 	OwnerLabelKey = "owner"
 )
 
+// TTL is a structure defining the TTL of a DNS record
+type TTL int64
+
+// IsConfigured returns true if TTL is configured, false otherwise
+func (ttl TTL) IsConfigured() bool {
+	return ttl > 0
```

Comment:
@linki The main reason I did this is to preserve the intent of the client at the `Endpoint` level. A not-configured TTL isn't the same as `300` for all providers. It should signal to the providers that the client did not specify a TTL and thus use the default, whatever the default might be for the given provider.

Not all providers have their default TTL set to 300. e.g., for CloudFlare, its default is (though not explicitly stated) ["automatic"](https://github.com/kubernetes-incubator/external-dns/blob/master/provider/cloudflare.go#L287). Having a way to distinguish b/w "configured" and "not configured, so use default" is useful IMHO.

What do you think? We can certainly make the case that TTL=300 is a sane default for all providers,  and if the client doesn't specify a custom TTL, it will get set to `300`.

Let me know. I don't mind changing the code.